backend/pkg/api/data_access/notifications.go

```diff
@@ -1639,47 +1640,47 @@ func (d *DataAccessService) UpdateNotificationSettingsNetworks(ctx context.Conte
 	}
 	return nil
 }
-func (d *DataAccessService) UpdateNotificationSettingsPairedDevice(ctx context.Context, userId uint64, pairedDeviceId uint64, name string, IsNotificationsEnabled bool) error {
-	result, err := d.userWriter.ExecContext(ctx, `
+
+func (d *DataAccessService) GetPairedDeviceUserId(ctx context.Context, pairedDeviceId uint64) (uint64, error) {
+	var userId uint64
+	err := d.userReader.GetContext(context.Background(), &userId, `
+		SELECT user_id
+		FROM users_devices
+		WHERE id = $1`, pairedDeviceId)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return 0, fmt.Errorf("%w, paired device with id %v not found", ErrNotFound, pairedDeviceId)
+		}
+		return 0, err
+	}
+	return userId, nil
+}
+
+func (d *DataAccessService) UpdateNotificationSettingsPairedDevice(ctx context.Context, pairedDeviceId uint64, name string, IsNotificationsEnabled bool) error {
+	_, err := d.userWriter.ExecContext(ctx, `
 		UPDATE users_devices 
 		SET 
 			device_name = $1,
 			notify_enabled = $2
-		WHERE user_id = $3 AND id = $4`,
-		name, IsNotificationsEnabled, userId, pairedDeviceId)
-	if err != nil {
-		return err
-	}
-
-	// TODO: This can be deleted when the API layer has an improved check for the device id
-	rowsAffected, err := result.RowsAffected()
-	if err != nil {
-		return err
-	}
-	if rowsAffected == 0 {
-		return fmt.Errorf("device with id %v to update notification settings not found", pairedDeviceId)
+		WHERE id = $3`,
+		name, IsNotificationsEnabled, pairedDeviceId)
+	if err == sql.ErrNoRows {
```

Comment:
**issue:** `ExecContext` does not return a `sql.ErrNoRows` if no rows were affected by either UPDATE or DELETE.

Reply:
fixed, ptal